jetstream/analysis.py

```diff
@@ -100,10 +100,15 @@ def _get_timelimits_if_ready(
         ):
             return None
 
+        # some experiments do not have and end date set yet
+        # set an arbitrary end date that is some time in the future
+        end_date = self.config.experiment.end_date
+        if end_date is None:
+            dates_enrollment = self.config.experiment.proposed_enrollment + 1
+            end_date = current_date + timedelta(days=dates_enrollment + 1)
```

Comment:
Since the `end_date` is required for calculating the `analysis_length_dates` it needs to be set to some future date which is not the current analysis date to prevent the overall analysis from running.

Reply:
Could we just return `None` instead of proposing a fake date?

Reply:
I'm not sure I entirely understand what you mean by that. We need some analysis window length for `TimeLimits.for_single_analysis_window` and having `_get_timelimits_if_ready` return `None` would skip the experiment analysis.

Reply:
I think it should only skip the Overall analysis, which is what we want, because it hasn't ended yet.

Reply:
Yeah, though `calculate_metrics` also expects some non-null `TimeLimit` to calculate the metrics for daily and weekly runs.

Reply:
see line 96, though; we only hit this code path if we're evaluating the overall window

Reply:
ah, I see!